pulser-core/pulser/noise_model.py

```diff
@@ -416,11 +409,9 @@ def _check_eff_noise(
             if operator.ndim != 2:
                 raise ValueError(f"Operator '{op!r}' is not a 2D array.")
 
-            if operator.shape not in possible_shapes:
-                raise ValueError(
-                    f"With{'' if with_leakage else 'out'} leakage, operator's "
-                    f"shape must be {possible_shapes[0]} or "
-                    f"{possible_shapes[1]}, not {operator.shape}."
+            if operator.shape != (2, 2):
+                raise NotImplementedError(
+                    f"Operator's shape must be (2,2) not {operator.shape}."
```

Comment:
I think you can still allow (3, 3) if `with_leakage=True` (it will then hit the `NotImplementedError` on the hamiltonian but that's a different story), just not (4,4)

Reply:
Yeah I agree,  I will do it this way :)

Reply:
I have decided to go for something a bit more clever: raise an error depending on if the shape might be implemented some day or not (assuming that we might never support shapes outside (2, 2), (3, 3) without leakage, and (3, 3), (4, 4) with leakage.